File: arven-towns-api/src/main/java/arvenwood/towns/api/resident/ResidentService.java
package arvenwood.towns.api.resident;

import org.spongepowered.api.Sponge;
import org.spongepowered.api.entity.living.player.Player;

import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface ResidentService {

    static ResidentService get() {
        return Sponge.getServiceManager().provideUnchecked(ResidentService.class);
    }

    Collection<Resident> getResidents();

    Optional<Resident> getResident(UUID uniqueId);

    Optional<Resident> getResident(String name);

    Resident getOrCreateResident(Player player);
}

File: src/main/kotlin/arvenwood/towns/api/resident/ResidentService.kt
package arvenwood.towns.api.resident

import org.spongepowered.api.Sponge
import org.spongepowered.api.entity.living.player.Player
import java.util.*

interface ResidentService {

    companion object {
        @JvmStatic
        fun get(): ResidentService =
            Sponge.getServiceManager().provideUnchecked(ResidentService::class.java)
    }

    val residents: Collection<Resident>

    fun getResident(uniqueId: UUID): Resident?

    fun getResident(name: String): Resident?

    fun getOrCreateResident(player: Player): Resident
}